pkg/kube/object.go

```diff
@@ -445,6 +445,59 @@ func (o *ObjectResolver) GetRelatedReplicasetName(ctx context.Context, object Ob
 	return "", fmt.Errorf("can only get related ReplicaSet for Deployment or Pod, not %q", string(object.Kind))
 }
 
+// GetNodeName will return the nodename from one of the running pod of any kubernetes kind
+func (o *ObjectResolver) GetNodeName(ctx context.Context, obj client.Object) (nodeName string, err error) {
+	switch t := obj.(type) {
+	case *corev1.Pod:
+		return (obj.(*corev1.Pod)).Spec.NodeName, nil
+	case *appsv1.Deployment:
+		replicaSet, err := o.ReplicaSetByDeployment(ctx, obj.(*appsv1.Deployment))
+		if err != nil {
+			return "", err
+		}
+		pods, err := o.getPodsByLabelSelector(ctx, obj.GetNamespace(), replicaSet.Spec.Selector.MatchLabels)
+		if err != nil || (pods != nil && len(pods.Items) == 0) {
```

Comment:
I think we should distinguish error returned by client-go (due to connection problems with K8s API server) and error returned because replicas count is 0.

And actually, I'm not sure that we should throw an error in case there are no pods.

Reply:
I checked on this, in getPodsbyLabelSelector method we use list method from client, so when there are no records matching those labels then we dont get any error and pods.Items is empty slice. so error is nil and len(pods.Items) will be zero. So we are not throwing any error when there are no pods. 
I think thats why I added a check with pods != nil && len(pods.Items) == 0, so that we will not get nil pointer error.

btw this will be needed for err check after ReplicaSetByDeployment(), so we will not throw error when there are no replicasets.